.github/workflows/gh-pages.yml

```
@@ -0,0 +1,83 @@
+name: "Publish to GitHub Pages"
+on:
+  workflow_dispatch:
+  release:
+    types:
+      - published
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+env:
+  GODOT_VERSION: 4.3
+
+jobs:
+  check:
+    name: Check if GitHub Pages is enabled
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check
+        id: check
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          if gh api "repos/${{ github.repository }}/pages" | jq --exit-status '.build_type == "workflow"'
+          then
+              echo "enabled=true" >> "$GITHUB_OUTPUT"
+          else
+              echo "# Not published to GitHub Pages" >> "$GITHUB_STEP_SUMMARY"
+              echo "" >> "$GITHUB_STEP_SUMMARY"
+              echo -n "Check that Pages is enabled, with the source set to GitHub Actions, in the " >> "$GITHUB_STEP_SUMMARY"
+              echo "[repository settings](https://github.com/${{ github.repository }}/settings/pages)." >> "$GITHUB_STEP_SUMMARY"
+          fi
+    outputs:
+      enabled: ${{ steps.check.outputs.enabled }}
+
+  build:
+    name: Build for web
+    needs:
+      - check
+    if: ${{ needs.check.outputs.enabled }}
+    runs-on: ubuntu-latest
+    container:
+      image: barichello/godot-ci:4.3
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          lfs: true
+
+      - name: Set up export templates
+        run: |
+          mkdir -v -p ~/.local/share/godot/export_templates/
+          mv /root/.local/share/godot/export_templates/${GODOT_VERSION}.stable ~/.local/share/godot/export_templates/${GODOT_VERSION}.stable
+
+      - name: Web Build
+        run: |
+          mkdir -v -p build/web
+          godot --headless --verbose --export-release "Web" ./build/web/index.html
+
+      - name: Upload Artifact
+        uses: actions/upload-pages-artifact@v3
+        with:
+          name: web
+          path: build/web
+
+  publish:
+    name: Publish to GitHub Pages
+    needs:
+      - build-web
```

Does this mean need the build's artifact web?

---

It means it needs the `build:` job, defined above, to have run successfully.

(Logically, yes, that's because it needs to use the `web` artefact that that job uploads.)

---

Argh. I should have paid more attention to your question. There is a typo here: this should read:

```suggestion
      - build
```